pallets/creditcoin/src/lib.rs

```diff
@@ -759,7 +765,7 @@ pub mod pallet {
 		pub fn add_ask_order(
 			origin: OriginFor<T>,
 			address_id: AddressId<T::Hash>,
-			terms: LoanTerms,
+			terms: LoanTerms<T::Hash>,
 			expiration_block: BlockNumberFor<T>,
 			guid: Guid,
 		) -> DispatchResult {
```

Comment:
should we have checks in `add_ask_order` and `add_bid_order` to make sure `address_id`'s blockchain matches the currency in `terms`?